CRM/Civicase/Hook/PostProcess/CaseCategoryPostProcessor.php

```diff
@@ -25,53 +23,28 @@ public function run($formName, CRM_Core_Form &$form) {
       return;
     }
 
-    $caseCategoryMenu = new CaseCategoryMenuService();
-    $caseCategoryCustomData = new CaseCategoryCustomDataType();
-    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
-
+    // Get object data from submitted from.
     $formValues = $form->_submitValues;
-    $formAction = $form->getVar('_action');
+    $caseCategoryValues = $form->getVar('_values');
+    $category = [
+      'id' => $form->getVar('_id'),
+      'name' => !empty($caseCategoryValues['name']) ? $caseCategoryValues['name'] : $formValues['label'],
+      'is_active' => $formValues['is_active'],
+      'icon' => $formValues['icon'],
+    ];
 
-    if ($formAction == CRM_Core_Action::DELETE) {
-      $caseCategoryValues = $form->getVar('_values');
-      $caseCategoryMenu->deleteItems($caseCategoryValues['name']);
-      $caseCategoryCustomFieldExtends->delete($caseCategoryValues['name']);
-      $caseCategoryCustomData->delete($caseCategoryValues['name']);
-    }
+    $op = $form->getVar('_action');
```

Comment:
Can we name this variable to be descriptive like it was before?
e.g `formAction` or something even better